pppd/plugins/pppoatm/misc.c

```diff
@@ -37,7 +37,7 @@ int __atmlib_fetch(const char **pos,...)
 	}
     }
     va_end(ap);
-    if (best > -1) (*pos) += best_len;
+    (*pos) += best_len;
```

Comment:
This is a semantic change; previously the function would return -1 if the string at *pos didn't match any of the argument strings at all, whereas now it will return 0, which callers interpret as a match with the second argument. So now text2qos will parse things incorrectly.

This is an example of why I would have preferred this commit to be split into smaller pieces, for example, four separate commits for chat, pppd, and the pppoatm and pppoe plugins. If you had, I could merge the commits that are OK. As it is, I can't merge any of it.